test/index.spec.js

```diff
@@ -226,6 +226,35 @@ describe('Redux promise middleware:', () => {
     });
 
     /**
+     * If the dispatch() call fails when dispatching the _FULFILLED action
+     * (for example, errors in connected component renders()), don't change it
+     * to a promise rejection
+     */
+    it('does not send a rejected action if disaptch fails', done => {
+      const fulfilledType = `${promiseAction.type}_FULFILLED`;
+      store = makeStore(undefined, (state, action) => {
+        if (action.type === fulfilledType) {
+          throw new Error();
+        } else {
+          return null;
+        }
+      });
+      const actionDispatched = store.dispatch(promiseAction);
+
+      actionDispatched.then(
+        ({ value, action }) => {
```

Comment:
The way you're doing `done` in tests works fine for tests that pass, but when tests fail due to the promise rejecting when it should resolve or resolving when it should reject, it doesn't encounter a `done` and waits for a timeout. The failure message of the test is that a the timeout was hit rather than our `expect(true).to.equal(false)`, so this is a better way to ensure the test provides helpful messages on failure.

This goes hand-in-hand with line 254, which forces the test to end when the promise resolves regardless of which branch it takes.